src/index.js

```diff
@@ -95,7 +98,7 @@ function isValidStaticPropertyName(name) {
 }
 
 function isValidRegex(pattern, flags) {
```

Comment:
The call site for this function (line 354) accesses a `flags` field that doesn't exist on the LiteralRegExpExpression node. I believe we replaced it in the es2016 version of the spec with separate fields for each flag, as in your tests below. Make sure the flags are being passed around appropriately, and add a test that would have caught this failure.